collect_app/src/main/java/org/odk/collect/android/preferences/dialogs/ResetDialogPreferenceFragmentCompat.kt

```diff
@@ -0,0 +1,270 @@
+        }
+        instances = view.findViewById<AppCompatCheckBox>(R.id.instances).apply {
+            setOnCheckedChangeListener(this@ResetDialogPreferenceFragmentCompat)
+        }
+        forms = view.findViewById<AppCompatCheckBox>(R.id.forms).apply {
+            setOnCheckedChangeListener(this@ResetDialogPreferenceFragmentCompat)
+        }
+        layers = view.findViewById<AppCompatCheckBox>(R.id.layers).apply {
+            setOnCheckedChangeListener(this@ResetDialogPreferenceFragmentCompat)
+        }
+        cache = view.findViewById<AppCompatCheckBox>(R.id.cache).apply {
+            setOnCheckedChangeListener(this@ResetDialogPreferenceFragmentCompat)
+        }
+        super.onBindDialogView(view)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        adjustResetButtonAccessibility()
+    }
+
+    override fun onDetach() {
+        preferences = null
+        instances = null
+        forms = null
+        layers = null
+        cache = null
+        super.onDetach()
+    }
+
+    override fun onClick(dialog: DialogInterface, which: Int) {
+        if (which == DialogInterface.BUTTON_POSITIVE) {
+            resetSelected()
+        }
+    }
+
+    override fun onDialogClosed(positiveResult: Boolean) {}
+
+    private fun resetSelected() {
+        val resetActions: MutableList<Int> = ArrayList()
+        if (preferences!!.isChecked) {
+            resetActions.add(RESET_PREFERENCES)
+        }
+        if (instances!!.isChecked) {
+            resetActions.add(ProjectResetter.ResetAction.RESET_INSTANCES)
+        }
+        if (forms!!.isChecked) {
+            resetActions.add(ProjectResetter.ResetAction.RESET_FORMS)
+        }
+        if (layers!!.isChecked) {
+            resetActions.add(ProjectResetter.ResetAction.RESET_LAYERS)
+        }
+        if (cache!!.isChecked) {
+            resetActions.add(ProjectResetter.ResetAction.RESET_CACHE)
+        }
+        if (resetActions.isNotEmpty()) {
+            showIfNotShowing(
+                ResetProgressDialog::class.java,
+                (_context as CollectAbstractActivity).supportFragmentManager
+            )
+
+            scheduler.immediate(
+                {
+                    return@immediate projectResetter.reset(resetActions)
+                },
+                { failedResetActions: List<Int> ->
+                    dismissDialog(
+                        ResetProgressDialog::class.java,
+                        (_context as CollectAbstractActivity).supportFragmentManager
+                    )
+                    handleResult(resetActions, failedResetActions)
+                }
+            )
+        }
+    }
+
+    private fun handleResult(resetActions: List<Int>, failedResetActions: List<Int>) {
+        val resultMessage = StringBuilder()
+        for (action in resetActions) {
+            when (action) {
+                RESET_PREFERENCES -> if (failedResetActions.contains(action)) {
```

Comment:
This builder is soo verbose. I think we could make it shorter by converting every case like (example for the first one):

```
                RESET_PREFERENCES -> resultMessage.append(
                    requireContext().getLocalizedString(
                        R.string.reset_settings_result,
                        if (failedResetActions.contains(action)) R.string.error_occured else R.string.success
                    )
                )
```

Reply:
Done.